projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageDeployerV2.java

```diff
@@ -0,0 +1,376 @@
+   *     caller is responsible for sending on success or in case of an exception)
+   * @param lastDeployedBy the name of the user that last deployed the data job
+   * @return true if it is successful and false if not.
+   */
+  public boolean scheduleJob(
+      DataJob dataJob,
+      JobDeployment jobDeployment,
+      Boolean sendNotification,
+      String lastDeployedBy,
+      Set<String> cronJobs) {
+    Validate.notNull(jobDeployment, "jobDeployment should not be null");
+    Validate.notNull(jobDeployment.getImageName(), "Image name is expected in jobDeployment");
+    log.info("Update cron job for data job {}", dataJob.getName());
+
+    try {
+      updateCronJob(dataJob, jobDeployment, lastDeployedBy, cronJobs);
+      return true;
+    } catch (ApiException e) {
+      return catchApiException(dataJob, jobDeployment, sendNotification, e);
+    }
+  }
+
+  private boolean catchApiException(
+      DataJob dataJob,
+      JobDeployment jobDeployment,
+      Boolean sendNotification,
+      ApiException apiException) {
+    if (apiException.getCode() == 422) {
+      try {
+        Gson gson = new Gson();
+        String msg =
+            NotificationContent.getErrorBody(
+                "Tried to deploy a data job",
+                "There has been an error in the configuration of your data job.",
+                "Your new/updated job was not deployed. Your job will run its latest successfully"
+                    + " deployed version (if any) as scheduled.",
+                "Please fix the job's configuration");
+        Map<Object, Object> error = gson.fromJson(apiException.getResponseBody(), HashMap.class);
+        if (error != null) {
+          log.error(
+              "Failed to schedule job due to Kubernetes client error (422). Generally input"
+                  + " validation should be done earlier. If this exception happens, most likely"
+                  + " we need to do some better input validation at client library. We are"
+                  + " assuming all k8s client error when creating cron job can be only customer"
+                  + " misconfiguration sending notification and reporting as user error",
+              apiException);
+          msg =
+              NotificationContent.getErrorBody(
+                  "Tried to deploy a data job",
+                  "There has been an error in the configuration of your data job : "
+                      + error.get("message"),
+                  "Your new/updated job was not deployed. Your job will run its latest"
+                      + " successfully deployed version (if any) as scheduled.",
+                  "Please fix the job's configuration");
+          log.error(msg);
+        }
+        deploymentProgress.failed(dataJob, DeploymentStatus.USER_ERROR, msg, sendNotification);
+        return false;
+      } catch (Exception ignored) {
+        log.debug("Failed to parse ApiException body, re-throwing it.: ", ignored);
```

Comment:
This log statement is misleading. It states that the exception is re-thrown, while in fact it is swallowed.

Reply:
Changed the message.